src/script.js

```diff
@@ -117,46 +137,66 @@ var gianniAccordion = (function(){
       this.transitionendevent = this.transitionendEventName();
       this.oneAtATime = true;
       this.selectedClass = "selected";
+      this.expandedClass = "expanded";
+      this.collapsedClass = "collapsed";
       this.trigger = "[data-accordion-element-trigger]";
       this.content = "[data-accordion-element-content]";
+      this.collapsedHeight = '0px';
+      this.defaultElements = ['a', 'button', 'input'];
+      this.openAtLandingIndex = -1
     }
 
-    setCustomData(data){
+    setCustomData(data) {
       var keys = Object.keys(data);
 
-      for(var i=0; i<keys.length; i++){
-        this[ keys[i] ] = data[ keys[i] ];
+      for (var i = 0; i < keys.length; i++) {
+        this[keys[i]] = data[keys[i]];
       }
     }
 
-    fire(eventName, el){
+    setLandinData(el) {
+      el.classList.add(this.collapsedClass);
+      el.querySelector(this.trigger).setAttribute('aria-expanded', 'false')
+      el.querySelector(this.content).setAttribute('aria-hidden', 'true')
+    }
+
+    fire(eventName, el) {
       var callbacks = this.events[eventName];
-      for(var i=0; i<callbacks.length; i++){
-        callbacks[i]( el )
+      for (var i = 0; i < callbacks.length; i++) {
+        callbacks[i](el)
       }
     }
 
-    on(eventName, cb){
-      if( !this.events[eventName] ) return       
+    on(eventName, cb) {
+      if (!this.events[eventName]) return
       this.events[eventName].push(cb);
     }
 
-    constructor(data){    
+    constructor(data) {
       this.setDefaultData();
       this.setCustomData(data); // ES6 => Object.assign(this, data)
-
-      [].forEach.call( document.querySelectorAll(this.elements), (el, i)=>{
+      [].forEach.call(document.querySelectorAll(this.elements), (el, i) => {
         this.els.push({
           wrapper: el,
           trigger: el.querySelector(this.trigger),
           content: el.querySelector(this.content)
         });
-
-        this.els[i].content.style.height = 0;
+        const element = this.els[i]
+        if (this.openAtLandingIndex !== i) {
+          element.content.style.height = this.collapsedHeight
+          this.setLandinData(el)
+        } else {
+          element.wrapper.classList.add(this.selectedClass)
```

Comment:
nell'else non si può chiamare direttamente this.open(element)? fa già tutta sta roba qua